addons/mail/models/mail_message_reaction.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class MailMessageReaction(models.Model):
+    _name = 'mail.message.reaction'
+    _description = 'Message Reaction'
+    _order = 'id desc'
+
+    message_id = fields.Many2one('mail.message', 'Message', ondelete='cascade', required=True, readonly=1)
```

Comment:
Hello.

I'm guessing we expect to have a ton of reaction records over time, correct?
So a couple things to consider:
- Having an index on "message_id", as this is probably the most used field when searching for reactions (could be wrong here, correct me if it's the case)
- Maybe remove the magic fields (create_uid, create_date, write_uid, write_date) with `_log_access = False` to save some database space?
- Having some kind of garbage collection mechanism to remove very old reactions to messages (not sure on that one, but I'm thinking old reactions are not really valuable)

Side note: feel free to disregard this comment and/or ignore me, I know this is not really my place.

Reply:
> Having an index on "message_id", as this is probably the most used field when searching for reactions (could be wrong here, correct me if it's the case)

There is already an index on message_id because of the unique constraint lower in the file. Might be worth a comment though.